test/odr_test.py

```diff
@@ -0,0 +1,48 @@
+import os
+import sys
+import re
+
+this_path = os.path.dirname(__file__)
+
+all_headers = set()
+include_path = this_path + "/../include/"
+for root, dirs, files in os.walk(include_path):
+    for fn in files:
+        fn = root + "/" + fn
+        assert fn.startswith(include_path)
+        fn = fn[len(include_path) :]
+        all_headers.add(fn)
+
+
+def get_headers(filename):
+    with open(filename) as f:
+        for hdr in re.findall('^#include [<"]([^>"]+)[>"]', f.read(), re.MULTILINE):
+            if not hdr.startswith("boost/histogram"):
+                continue
+            yield hdr
+
+
+included_headers = set()
+unread_headers = set()
+for hdr in get_headers(this_path + "/odr_test.cpp"):
```

Comment:
```suggestion
for hdr in get_headers(os.path.join(this_path, "/odr_test.cpp")):
```

This is why I love pathlib, plumbum.path, pretty much anything OO for paths!

Reply:
I don't mind using os.path.join for the time being.